src/reducers/user_reducer.js

```diff
@@ -0,0 +1,39 @@
+import {
+  LOGIN_BEGIN,
+  LOGIN_ERROR,
+  LOGIN_SUCCESS,
+  LOGOUT_BEGIN,
+  LOGOUT_ERROR,
+  LOGOUT_SUCCESS,
+  REGISTER_BEGIN,
+  REGISTER_ERROR,
+  REGISTER_SUCCESS,
+} from '../actions';
+
+const user_reducer = (state, action) => {
```

Comment:
React will call the reducer function anytime we do a `dispatch(...)` call. The function will get the current state, and the value of whatever we put inside `dispatch`. We can use this to essentially "listen" for events and change state values however we want here